core/src/main/java/nl/nn/adapterframework/webcontrol/api/SendJmsMessage.java

```diff
@@ -57,55 +54,47 @@
 	@Path("jms/message")
 	@Produces(MediaType.APPLICATION_JSON)
 	@Consumes(MediaType.MULTIPART_FORM_DATA)
-	public Response putJmsMessage(MultipartFormDataInput input) throws ApiException {
+	public Response putJmsMessage(MultipartBody inputDataMap) throws ApiException {
 
 		String jmsRealm = null, destinationName = null, destinationType = null, replyTo = null, message = null, fileName = null;
 		InputStream file = null;
 		boolean persistent = false;
-		Map<String, List<InputPart>> inputDataMap = input.getFormDataMap();
+		
 		if(inputDataMap == null) {
 			throw new ApiException("Missing post parameters");
 		}
 
 		try {
-			if(inputDataMap.get("realm") != null)
-				jmsRealm = inputDataMap.get("realm").get(0).getBodyAsString();
+			if(inputDataMap.getAttachmentObject("real", String.class) != null)
+				jmsRealm = inputDataMap.getAttachmentObject("real", String.class);
 			else
 				throw new ApiException("JMS realm not defined", 400);
-			if(inputDataMap.get("destination") != null)
-				destinationName = inputDataMap.get("destination").get(0).getBodyAsString();
+			if(inputDataMap.getAttachmentObject("destination", String.class) != null)
```

Comment:
Maybe instead of all these ugly if else statements you could use:
resolveStringFromMap(inputDataMap, "destination");
This will, indirectly, have the same result and will clean up the code dramatically!